### PR TITLE
Added ability for Java 9+ clients to utilize anonymous object methods

### PR DESCRIPTION
**Pull request by Spinsie**
#93

### CODE REVIEW

File: src/main/java/horseshoe/Settings.java [hunk 1]
```diff
@@ -166,30 +177,21 @@ public String escape(final String raw) {
...
-					"java.util.stream.Collectors")) {
-				try {
-					defaultLoadableClasses.add(Class.forName(className));
-				} catch (final ClassNotFoundException e) {
-					Template.LOGGER.log(Level.WARNING, "Failed to load class \"{0}\", even though Java 8 or later was detected.", className);
-				}
-			}
-		}
+				Pattern.class,
+
```
Comment (nicklauslittle) [inline src/main/java/horseshoe/Settings.java:182]: You missed Duration. Also, Thomas wanted String (for `instanceof()`).

File: src/main/java/horseshoe/internal/MethodHandler.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	private static final LookerUpper lookerUpper = Utilities.loadJavaSpecialization(javaVersion -> javaVersion >= 9.0 ? MethodHandler.class.getName() + "$PrivateLookerUpper" : defaultLookerUpper);
+
+	private interface LookerUpper {
+		Lookup lookup(Class<?> targetClass) throws IllegalAccessException;
+	}
+
+	@SuppressWarnings("unused")
+	private static class PrivateLookerUpper implements LookerUpper {
+
+		public PrivateLookerUpper() { }
```
Comment (nicklauslittle) [inline src/main/java/horseshoe/internal/MethodHandler.java]: Remove empty public constructor.

File: src/test/java/horseshoe/internal/MethodHandlerTests.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import java.io.StringWriter;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicReference;
+
+import horseshoe.Template;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+
+public class MethodHandlerTests {
```
Comment (nicklauslittle) [inline src/test/java/horseshoe/internal/MethodHandlerTests.java]: Remove public

File: build.gradle [hunk 4]
```diff
@@ -17,19 +17,18 @@ checkstyle {
 }
 
 java {
-	sourceCompatibility = JavaVersion.VERSION_1_8
-	targetCompatibility = JavaVersion.VERSION_1_8
+	sourceCompatibility = JavaVersion.VERSION_1_9
+	targetCompatibility = JavaVersion.VERSION_1_9
```
Comment (nicklauslittle) [inline build.gradle]: I'm not sure I want to require Java 9 for development. Maybe we can just remove this section?

File: src/main/java/horseshoe/Settings.java (hunk 1 above)
Comment (Spinsie) [reply src/main/java/horseshoe/Settings.java:182]: I added Duration. String is already present.

File: src/main/java/horseshoe/internal/MethodHandler.java (hunk 2 above)
Comment (Spinsie) [reply src/main/java/horseshoe/internal/MethodHandler.java]: Done

File: src/test/java/horseshoe/internal/MethodHandlerTests.java (hunk 3 above)
Comment (Spinsie) [reply src/test/java/horseshoe/internal/MethodHandlerTests.java]: Done

File: build.gradle (hunk 4 above)
Comment (Spinsie) [reply build.gradle]: Done

**Comment by nicklauslittle** [comment]
The test appears to fail on Java 8. I tried changing the test so that the method is private, but then it fails on Java 11. Can we change the test so that the `privateLookupIn()` method is used appropriately in the test?

**Comment by Spinsie** [comment]
> The test appears to fail on Java 8. I tried changing the test so that the method is private, but then it fails on Java 11. Can we change the test so that the `privateLookupIn()` method is used appropriately in the test?

I am not sure why, but I reverted the code for `privateLookupIn()` entirely and now everything works. Maybe the java 8 refactor did the trick.

**Pull request merged**